runtime/src/snapshot_package.rs

```diff
@@ -50,41 +52,67 @@ impl SnapshotPackage {
         }
 
         let bank_fields_to_serialize = bank.get_fields_to_serialize();
-        Self {
-            snapshot_kind,
-            slot,
-            block_height: bank.block_height(),
-            hash: SnapshotHash::new(bank_fields_to_serialize.accounts_lt_hash.0.checksum()),
-            snapshot_storages,
-            status_cache_slot_deltas,
-            bank_fields_to_serialize,
+
+        let snapshot_archive_package = match snapshot_kind {
+            SnapshotKind::Archive(snapshot_archive_kind) => {
+                let checksum = bank_fields_to_serialize.accounts_lt_hash.0.checksum();
+                Some(SnapshotArchivePackage {
+                    snapshot_archive_kind,
+                    hash: SnapshotHash::new(checksum),
+                })
+            }
+        };
+
+        let bank_snapshot_package = BankSnapshotPackage {
+            bank_fields: bank_fields_to_serialize,
             bank_hash_stats: bank.get_bank_hash_stats(),
+            slot_deltas: status_cache_slot_deltas,
             write_version: bank
                 .rc
                 .accounts
                 .accounts_db
                 .write_version
                 .load(Ordering::Acquire),
+        };
+
+        Self {
+            snapshot_kind,
+            block_height: bank.block_height(),
+            slot,
+            bank_snapshot_package,
+            snapshot_archive_package,
+            snapshot_storages,
             enqueued: Instant::now(),
         }
     }
 }
 
+#[cfg(feature = "dev-context-only-utils")]
+use crate::bank::{BankFieldsToSerialize, BankHashStats};
```

Comment:
Interesting, I'm surprised it didn't get flagged/moved automatically. Whatever the case, can this be moved to the top, with the other imports, please?